src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -3114,10 +3114,16 @@ private void ImportLdToken(int token)
             if (ldtokenValue is TypeDesc)
             {
                 ldtokenKind = WellKnownType.RuntimeTypeHandle;
-                PushLoadExpression(StackValueKind.ByRef, "ldtoken", GetEETypePointerForTypeDesc(ldtokenValue as TypeDesc, false), _compilation.TypeSystemContext.SystemModule.GetKnownType("System", "EETypePtr"));
+                var typeDesc = (TypeDesc)ldtokenValue;
+                PushLoadExpression(StackValueKind.ByRef, "ldtoken", GetEETypePointerForTypeDesc(typeDesc, false), _compilation.TypeSystemContext.SystemModule.GetKnownType("System", "EETypePtr"));
                 MethodDesc helper = _compilation.TypeSystemContext.GetHelperEntryPoint("LdTokenHelpers", "GetRuntimeTypeHandle");
                 AddMethodReference(helper);
                 HandleCall(helper, helper.Signature);
+                if (ConstructedEETypeNode.CreationAllowed(typeDesc))
```

Comment:
This should rather be `MaximallyConstructableType` that does the right thing for both `CreationAllowed` and `!CreationAllowed` cases.

Reply:
Thanks, have changed to be unconditionally `MaximallyConstructableType` and removed couple of unused variables.